common/src/main/java/com/wynntils/screens/maps/SeaskipperMapScreen.java

```diff
@@ -0,0 +1,532 @@
+
+            FontRenderer.getInstance()
+                    .renderText(
+                            poseStack,
+                            CodedString.fromString("Inaccessible"),
+                            10,
+                            10 + renderYOffset,
+                            CommonColors.GRAY,
+                            HorizontalAlignment.LEFT,
+                            VerticalAlignment.TOP,
+                            TextShadow.OUTLINE);
+        }
+
+        FontRenderer.getInstance()
+                .renderAlignedTextInBox(
+                        poseStack,
+                        CodedString.fromString(hoveredPoi.getName()),
+                        7,
+                        textureWidth,
+                        Texture.TERRITORY_TOOLTIP_TOP.height() + centerHeight,
+                        Texture.TERRITORY_TOOLTIP_TOP.height() + centerHeight + Texture.TERRITORY_NAME_BOX.height(),
+                        0,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.LEFT,
+                        VerticalAlignment.MIDDLE,
+                        TextShadow.OUTLINE);
+
+        poseStack.popPose();
+    }
+
+    @Override
+    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+        for (GuiEventListener child : children()) {
+            if (child.isMouseOver(mouseX, mouseY)) {
+                child.mouseClicked(mouseX, mouseY, button);
+                return true;
+            }
+        }
+
+        for (SeaskipperPoi poi : seaskipperPois) {
+            if (poi.isSelected(mouseX, mouseY)) {
+                buyPass(poi.getName());
+
+                return true;
+            }
+        }
+
+        return super.mouseClicked(mouseX, mouseY, button);
+    }
+
+    public void setSeaskipperPois(List<SeaskipperPoi> seaskipperPois) {
+        this.seaskipperPois = seaskipperPois;
+    }
+
+    private void toggleBorders() {
+        hideTerritoryBorders = !hideTerritoryBorders;
+    }
+
+    private void toggleDestinations() {
+        renderAllDestinations = !renderAllDestinations;
+    }
+
+    private void toggleRoutes() {
+        renderRoutes = !renderRoutes;
+    }
+
+    private void buyBoat() {
+        ContainerUtils.clickOnSlot(
+                boatSlot,
+                actualSeaskipperScreen.getMenu().containerId,
+                GLFW.GLFW_MOUSE_BUTTON_LEFT,
+                actualSeaskipperScreen.getMenu().getItems());
+    }
+
+    private void buyPass(String destinationToTravelTo) {
+        int passSlot = -1;
+
+        for (Map.Entry<SeaskipperDestinationItem, Integer> entry : destinations.entrySet()) {
+            String destination = entry.getKey().getDestination();
+
+            if (destination.equals(destinationToTravelTo)) {
+                passSlot = entry.getValue();
+                break;
+            }
+        }
+
+        if (passSlot == -1) {
+            return;
+        }
+
+        ContainerUtils.clickOnSlot(
```

Comment:
Seems you can unify this part with the code in buyBoat().

Reply:
... and, if you chose to fix the model fully, move this functionality to the model instead.